modules/gapi/include/opencv2/gapi/video.hpp

```diff
@@ -129,6 +129,106 @@ G_TYPED_KERNEL(GBackgroundSubtractor, <GMat(GMat, BackgroundSubtractorParams)>,
+    {
+        GAPI_Assert(dp > 0 && mp > 0);
+        GAPI_Assert(depth == CV_32F || depth == CV_64F);
+        ctrlDim = std::max(cp, 0);
+        type = depth;
+        dpDim = dp;
+        mpDim = mp;
+
+        statePre = Mat::zeros(dpDim, 1, type);
+        transitionMatrix = Mat::eye(dpDim, dpDim, type);
+
+        processNoiseCov = Mat::eye(dpDim, dpDim, type);
+        measurementMatrix = Mat::zeros(mpDim, dpDim, type);
+        measurementNoiseCov = Mat::eye(mpDim, mpDim, type);
+
+        errorCovPre = Mat::zeros(dpDim, dpDim, type);
+
+        if (ctrlDim > 0)
+            controlMatrix = Mat::zeros(dpDim, ctrlDim, type);
+        else
+            controlMatrix.release();
+    }
+};
+
+G_TYPED_KERNEL(GKalmanFilter, <GMat(GMat, GOpaque<bool>, GMat, KalmanParams)>, "org.opencv.video.KalmanFilter")
+{
+    static GMatDesc outMeta(const GMatDesc& measurement, const GOpaqueDesc&, const GMatDesc&, const KalmanParams& kfParams)
+    {
+        GAPI_Assert(kfParams.type == CV_32F || kfParams.type == CV_64F);
+        GAPI_Assert(kfParams.dpDim > 0 && kfParams.mpDim > 0 && kfParams.ctrlDim >= 0);
+
+        GAPI_Assert(kfParams.statePre.type() == kfParams.type &&
+                    kfParams.errorCovPre.type() == kfParams.type &&
+                    kfParams.transitionMatrix.type() == kfParams.type &&
+                    kfParams.measurementMatrix.type() == kfParams.type &&
+                    kfParams.processNoiseCov.type() == kfParams.type &&
+                    kfParams.measurementNoiseCov.type() == kfParams.type);
+
+        if (!kfParams.controlMatrix.empty() && kfParams.ctrlDim > 0)
+            GAPI_Assert(kfParams.controlMatrix.type() == kfParams.type &&
+                        kfParams.controlMatrix.cols == kfParams.ctrlDim &&
+                        kfParams.controlMatrix.rows == kfParams.dpDim);
+
+        GAPI_Assert(!kfParams.statePre.empty() && !kfParams.errorCovPre.empty() &&
+                    !kfParams.transitionMatrix.empty() && !kfParams.measurementMatrix.empty() &&
+                    !kfParams.processNoiseCov.empty() && !kfParams.measurementNoiseCov.empty());
+
+        GAPI_Assert(kfParams.statePre.rows == kfParams.dpDim && kfParams.statePre.cols == 1);
+        GAPI_Assert(kfParams.measurementMatrix.cols == kfParams.dpDim &&
+                    kfParams.measurementMatrix.rows == kfParams.mpDim);
+        GAPI_Assert(kfParams.errorCovPre.rows == kfParams.errorCovPre.cols && kfParams.errorCovPre.rows == kfParams.dpDim);
+
+        GAPI_Assert((kfParams.transitionMatrix.rows == kfParams.transitionMatrix.cols) && kfParams.transitionMatrix.rows == kfParams.dpDim);
+        GAPI_Assert((kfParams.processNoiseCov.rows == kfParams.processNoiseCov.cols) && kfParams.processNoiseCov.rows == kfParams.dpDim);
+        GAPI_Assert((kfParams.measurementNoiseCov.rows == kfParams.measurementNoiseCov.cols) && kfParams.measurementNoiseCov.rows == kfParams.mpDim);
+
+        return measurement.withSize(Size(1, kfParams.dpDim));
```

Comment:
Should `measurement` be with the `kfParams.type` as well? I think ther should be an assert

Reply:
There should be call of function withDepth() I think.

Reply:
But wait, can measurements have another type? Shouldn't we check in `outMeta` what `Mat` was passed to the kernel? It is not only about the type, but also about the size. Could you please add the checks for `measurements`?

Reply:
Done